rustls/src/kx.rs

```diff
@@ -67,20 +65,19 @@ impl KeyExchange {
         self.skxg.name
     }
 
-    fn decode_client_params(&self, kx_params: &[u8]) -> Option<ClientECDHParams> {
+    fn decode_ecdh_params<T: Codec>(kx_params: &[u8]) -> Option<T> {
         let mut rd = Reader::init(kx_params);
-        let ecdh_params = ClientECDHParams::read(&mut rd).unwrap();
-        if rd.any_left() {
-            None
-        } else {
-            Some(ecdh_params)
+        let ecdh_params = T::read(&mut rd)?;
+        match rd.any_left() {
+            false => Some(ecdh_params),
+            true => None,
```

Comment:
I was expecting that codecov would report this line as uncovered by tests, but it actually reports it is covered already. I guess the test for this is somewhere in the server testing code and so maybe we should copy-pasta it, or otherwise refactor it, so it is testing the client too?

Reply:
I wonder, should we report these kinds of minor coverage gaps in bogo, too?

Reply:
IDK. The issue tracker for BoringSSL is https://bugs.chromium.org/p/boringssl/issues but I don't think filing an issue is very productive. Perhaps writing a PR for adding the test would be more effective, especially since Rustls is already using a fork (currently with no changes?) of BoGo.

Reply:
For now I just added some unit tests that cover #640 and #641.